src/components/AdminPage/CategoryList.js

```diff
@@ -2,67 +2,113 @@ import React, { Component } from "react";
+  componentDidUpdate() {
     const { selectedCategory } = this.state;
-    const index = selectedCategory.indexOf(selected);
-
-    // eslint-disable-next-line no-unused-expressions
-    index !== -1
-      ? selectedCategory.splice(index, 1)
-      : selectedCategory.push(selected);
-
-    const filteredResource = this.props.resource.filter(resource =>
-      this.state.selectedCategory.some(searchCategory =>
-        resource.categories
-          .split(",")
-          .map(item => item.trim())
-          .includes(searchCategory)
-      )
-    );
+    const { resource } = this.props;
+    const filteredResource = [];
+
+    if (selectedCategory.length === 0) {
+      this.props.actions.filterByCategories(resource);
+    } else {
+      resource.forEach(res => {
+        const isMatch = selectedCategory.some(cat => res.categories === cat);
 
-    this.props.actions.filterByCategories(
-      selectedCategory.length > 0 ? filteredResource : this.props.resource
+        if (isMatch) {
+          filteredResource.push(res);
+        }
+      });
+      this.props.actions.filterByCategories(filteredResource);
+    }
+  }
+
+  handleClick = async event => {
+    event.persist();
+    const isContains = event.target.classList.contains(
+      "list-group-item-success"
     );
+
+    const selectedCategoryLength = this.state.selectedCategory.length;
+
+    if (isContains && selectedCategoryLength === 1) {
+      this.setState({
+        selectedCategory: [],
+      });
+    } else if (isContains) {
+      this.setState(prevState => {
+        const selectedCategoryCopy = prevState.selectedCategory.slice();
+        _.remove(selectedCategoryCopy, cat => cat === event.target.innerHTML);
+
+        return {
+          selectedCategory: selectedCategoryCopy,
+        };
+      });
+    } else {
+      this.setState(prevState => ({
+        selectedCategory: [
+          ...prevState.selectedCategory,
+          event.target.innerHTML,
+        ],
+      }));
+    }
+  };
+
+  clearChecks = () => {
+    this.setState({
+      selectedCategory: [],
+    });
   };
 
   render() {
-    const categoryMenuItems = this.props.categories.map(cat => (
-      <FormGroup key={cat} check>
-        <Input
-          type="checkbox"
-          key={cat}
-          onChange={() => this.handleChange(cat)}
-        />
-        {cat}
-      </FormGroup>
+    const { selectedCategory } = this.state;
+    const { categories } = this.props;
+    categories.sort();
+    const categoryMenuItems = categories.map((curr, index) => (
+      <ListGroupItem
+        key={index.toString()}
+        color={_.indexOf(selectedCategory, curr) !== -1 ? "success" : ""}
+        onClick={this.handleClick}
+      >
+        {curr}
+      </ListGroupItem>
     ));
 
     return (
-      <Form>
+      <div>
         <Label>Filter by Category</Label>
-        {categoryMenuItems}
-      </Form>
+        <Button color="info" className="w-100" onClick={this.clearChecks}>
```

Comment:
Having this render below the categories makes a bit more sense on the UX perspective